pkg/controller/ec2/routetable/controller.go

```diff
@@ -237,33 +250,110 @@ func (e *external) Delete(ctx context.Context, mgd resource.Managed) error {
 	return awsclient.Wrap(resource.Ignore(ec2.IsRouteTableNotFoundErr, err), errDelete)
 }
 
-func (e *external) createRoutes(ctx context.Context, tableID string, desired []v1alpha4.Route, observed []v1alpha4.RouteState) error {
+func (e *external) deleteRoutes(ctx context.Context, tableID string, desired []v1alpha5.Route, observed []v1alpha5.RouteState) error { // nolint:gocyclo
+	for _, rt := range observed {
+		found := false
+		for _, ds := range desired {
+			if aws.StringValue(ds.DestinationCIDRBlock) == rt.DestinationCIDRBlock && (aws.StringValue(ds.GatewayID) == rt.GatewayID ||
+				aws.StringValue(ds.InstanceID) == rt.InstanceID ||
+				aws.StringValue(ds.LocalGatewayID) == rt.LocalGatewayID ||
+				aws.StringValue(ds.NatGatewayID) == rt.NatGatewayID ||
+				aws.StringValue(ds.NetworkInterfaceID) == rt.NetworkInterfaceID ||
+				aws.StringValue(ds.TransitGatewayID) == rt.TransitGatewayID ||
+				aws.StringValue(ds.VpcPeeringConnectionID) == rt.VpcPeeringConnectionID) {
+
+				found = true
+				break
+			}
+		}
+		if !found {
+			_, err := e.client.DeleteRouteRequest(&awsec2.DeleteRouteInput{
+				RouteTableId:             aws.String(tableID),
+				DestinationCidrBlock:     aws.String(rt.DestinationCIDRBlock),
+				DestinationIpv6CidrBlock: aws.String(rt.DestinationIPV6CIDRBlock),
+			}).Send(ctx)
+
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (e *external) createRoutes(ctx context.Context, tableID string, desired []v1alpha5.Route, observed []v1alpha5.RouteState) error { // nolint:gocyclo
 	for _, rt := range desired {
 		isObserved := false
 		for _, ob := range observed {
-			if ob.GatewayID == aws.StringValue(rt.GatewayID) && ob.DestinationCIDRBlock == aws.StringValue(rt.DestinationCIDRBlock) {
+			if ob.DestinationCIDRBlock == aws.StringValue(rt.DestinationCIDRBlock) && (ob.GatewayID == aws.StringValue(rt.GatewayID) ||
```

Comment:
I'm not sure about this condition. What happens when user changes only one or a few fields but not all? In such cases, we should delete and recreate the route since there is no update call but from what I can tell, having one field match is OK to skip doing any operation. Should we use `&&` instead of `||`?

Reply:
This is a fun one. Will try my best to explain but it should be checking that the destinationCIDR AND a destination is still the same. Since there should only be one destination per route ( which is why i was doing the `||`) I think `&&` should work though and better enforce the check will try it out.